src/dataflow/src/sink/kafka.rs

```diff
@@ -222,19 +274,61 @@ where
                 input.for_each(|_, rows| {
                     rows.swap(&mut vector);
 
-                    for (row, _time, diff) in vector.drain(..) {
-                        queue.push_back((row, diff));
+                    for (row, time, diff) in vector.drain(..) {
+                        queue.push_back((row, time.to_string(), diff));
+                        if let Some(consistency) = &mut consistency {
+                            // Note that since a single differential message
+                            // turns into |diff| messages we need to increment
+                            // message counts by |diff| instead of just 1 for
+                            // the consistency topic
+                            consistency.update_timestamp_count(time, diff.abs() as usize);
+                        }
                     }
                 });
 
+                if let Some(consistency) = &mut consistency {
+                    // Find the timestamps that are now complete (meaning all
+                    // timestamps t !<= input_frontier. For each consistency
+                    // send a pair of BEGIN / END messages in the consistency topic
+                    consistency
+                        .get_complete_timestamps(input.frontier())
+                        .iter()
+                        .for_each(|(k, v)| {
+                            println!("timestamp {} produced {} messages", k, v);
+                            let transaction_id = k.to_string();
+                            let begin = encoder.encode_transaction_unchecked(
```

Comment:
the way that we are writing out this data is kind of misleading. the consistency topic implies a series of transactions with no overlap, but we 100% will emit actual records out of transaction order. proposal: emit the begin message the first time we see a Row with that timestamp, and then emit a end message when we know we cannot see any more?